packages/vitest/src/node/core.ts

```diff
@@ -336,14 +336,14 @@ export class Vitest {
       if (hasFailed(files))
         process.exitCode = 1
 
-      if (!this.config.browser)
-        await this.report('onFinished', files, this.state.getUnhandledErrors())
       this.cache.results.updateResults(files)
       await this.cache.results.writeToCache()
     })()
-      .finally(() => {
+      .finally(async () => {
         this.runningPromise = undefined
```

Comment:
We should move `this.runningPromise` after `onFinished`

Reply:
good catch